phpBB/phpbb/storage/adapter/local.php

```diff
@@ -159,4 +161,21 @@ public function create_dir($path)
 		}
 	}
 
+	public function download($path, $filename)
```

Comment:
imo, this function should return the file path ($this->root_path . $path) and the Response be constructed by the controller.

And the method could be renamed to get_file_path() or something like that.